src/Item_Line.php

```diff
@@ -0,0 +1,576 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2022 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+class Item_Line extends CommonDBRelation
+{
+    public static $itemtype_1 = 'Line';
+    public static $items_id_1 = 'lines_id';
+    public static $itemtype_2 = 'itemtype';
+    public static $items_id_2 = 'items_id';
+    public static $checkItem_1_Rights = self::DONT_CHECK_ITEM_RIGHTS;
+    public static $mustBeAttached_1      = false;
+    public static $mustBeAttached_2      = false;
```

Comment:
Maybe, but these properties break the implementation. When opening the form to add a line to an asset, you get an empty form. I copied the code from Item_Cluster and made adjustments as needed and I suspect that feature had similar issues.

When displaying the form, `common.tabs.php` checks if you can create the item because the ID is not yet set. This checks both the static `canCreate` and `canCreateItem`. CommonDBRelation then decides during the `canCreateItem` check that the new, not created item doesn't yet have a line or item associated so it returns false. Therefore, you cannot display a new item form if you choose to enforce link validity using these properties.

Reply:
I changed forms to not use the `initForm` method, so I was able to use default rights checks. Is it OK for you ?